pandas/plotting/_matplotlib/hist.py

```diff
@@ -243,15 +244,26 @@ def _grouped_hist(
     sharey : bool, default False
     rot : int, default 90
     grid : bool, default True
+    legend: : bool, default False
     kwargs : dict, keyword arguments passed to matplotlib.Axes.hist
 
     Returns
     -------
     collection of Matplotlib Axes
     """
+    if legend:
+        if isinstance(data, ABCDataFrame):
+            if column is None:
+                kwargs["label"] = data.columns
+            else:
```

Comment:
I think you can just remove this branch

Reply:
When you say branch - do you mean just the else clause?

Reply:
Ah - perhaps you're thinking that in the case column is specified, then data only consists of those columns? I checked that this is not the case:

    if legend and data.ndim != 1 and column is not None:
        assert data.columns.tolist() == [column], f"{data.columns}, {column}"

gives

    AssertionError: Index(['a', 'b'], dtype='object'), b